sdktests/server_side_data_system_base.go

```diff
@@ -0,0 +1,112 @@
+package sdktests
+
+import (
+	"time"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/launchdarkly/go-sdk-common/v3/ldvalue"
+	"github.com/launchdarkly/go-server-sdk-evaluation/v3/ldbuilders"
+	"github.com/launchdarkly/sdk-test-harness/v2/framework/ldtest"
+	"github.com/launchdarkly/sdk-test-harness/v2/servicedef"
+	"github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/require"
+)
+
+func doServerSideDataSystemTests(t *ldtest.T) {
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     "localhost:6379",
+		Password: "", // no password set
+		DB:       0,  // use default DB
+	})
+
+	newServerSideDataSystemTests(t, RedisPersistenceStore{redis: rdb}).Run(t)
+}
+
+type ServerSideDataSystemTests struct {
+	persistence  PersistenceStore
+	initialFlags map[string]string
+}
+
+func newServerSideDataSystemTests(t *ldtest.T, persistence PersistenceStore) *ServerSideDataSystemTests {
+	flagKeyBytes, err :=
+		ldbuilders.NewFlagBuilder("flag-key").Version(1).
+			On(true).Variations(ldvalue.String("off"), ldvalue.String("match"), ldvalue.String("fallthrough")).
+			OffVariation(0).
+			FallthroughVariation(2).
+			Build().MarshalJSON()
+	require.NoError(t, err)
+
+	initialFlags := map[string]string{"flag-key": string(flagKeyBytes)}
+
+	uncachedFlagKeyBytes, err :=
+		ldbuilders.NewFlagBuilder("uncached-flag-key").Version(1).
+			On(true).Variations(ldvalue.String("off"), ldvalue.String("match"), ldvalue.String("fallthrough")).
+			OffVariation(0).
+			FallthroughVariation(2).
+			Build().MarshalJSON()
+	require.NoError(t, err)
+
+	initialFlags["uncached-flag-key"] = string(uncachedFlagKeyBytes)
```

Comment:
Just some initial data to play with so I don't have to define it every single time.